src/migrations/20231101171425-additional-changes-for-merge-goals.js

```diff
@@ -0,0 +1,59 @@
+const { prepMigration, addValuesToEnumIfTheyDontExist } = require('../lib/migration');
+const { GOAL_CREATED_VIA } = require('../constants');
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const sessionSig = __filename;
+      await prepMigration(queryInterface, transaction, sessionSig);
+
+      await addValuesToEnumIfTheyDontExist(queryInterface, 'enum_Goals_createdVia', GOAL_CREATED_VIA);
+
+      await queryInterface.addColumn('Objectives', 'mapsToParentObjectiveId', {
+        type: Sequelize.INTEGER,
+        allowNull: true,
+        default: null,
+        references: {
+          model: {
+            tableName: 'Objectives',
+          },
+          key: 'id',
+        },
+      }, { transaction });
+
+      await queryInterface.addColumn('ActivityReportObjectives', 'originalObjectiveId', {
+        type: Sequelize.INTEGER,
+        allowNull: true,
+        default: null,
+        references: {
+          model: {
+            tableName: 'Objectives',
+          },
+          key: 'id',
+        },
+      }, { transaction });
+
+      return queryInterface.addColumn('ActivityReportGoals', 'originalGoalId', {
+        type: Sequelize.INTEGER,
+        allowNull: true,
+        default: null,
+        references: {
+          model: {
+            tableName: 'Goals',
+          },
+          key: 'id',
+        },
+      }, { transaction });
+    });
+  },
+  async down(queryInterface) {
+    await queryInterface.sequelize.transaction(async (transaction) => {
+      const sessionSig = __filename;
```

Comment:
Can we remove the enum?

Reply:
[not supported](https://www.postgresql.org/docs/current/datatype-enum.html)